src/js/player.js

```diff
@@ -1570,20 +1570,24 @@ class Player extends Component {
   /**
    * start media playback
    *
-   * @return {Player}
-   *         A reference to the player object this function was called on
+   * @return {Promise|undefined}
+   *         Returns a `Promise` if the browser returns one, for most browsers this will
+   *         return undefined.
    */
   play() {
     // Only calls the tech's play if we already have a src loaded
     if (this.src() || this.currentSrc()) {
-      this.techCall_('play');
-    } else {
-      this.tech_.one('loadstart', function() {
-        this.play();
-      });
+      return this.techGet_('play');
     }
 
-    return this;
+    this.tech_.one('loadstart', function() {
```

Comment:
If there's a source but it's of an unknown type, this throws as `this.tech_` is undefined.